src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -266,8 +266,26 @@ private static int GetMinRequiredLength(Regex r)
         [InlineData("(?:a+){4}", "a{4,}")]
         [InlineData("(?:a{1,2}){4}", "a{4,8}")]
         // Alternation reduction
+        [InlineData("a|b", "[ab]")]
         [InlineData("a|b|c|d|e|g|h|z", "[a-eghz]")]
         [InlineData("a|b|c|def|g|h", "[a-c]|def|[gh]")]
+        [InlineData("this|that|there|then|those", "th(?:is|at|ere|en|ose)")]
+        [InlineData("it's (?>this|that|there|then|those)", "it's (?>th(?:is|at|ere|en|ose))")]
+        [InlineData("it's (?>this|that|there|then|those)!", "it's (?>th(?:is|at|ere|en|ose))!")]
+        [InlineData("abcd|abce", "abc[de]")]
+        [InlineData("abcd|abef", "ab(?:cd|ef)")]
+        [InlineData("abcd|aefg", "a(?:bcd|efg)")]
+        [InlineData("abcd|abc|ab|a", "a(?:bcd|bc|b|(?:))")]
```

Comment:
Is `a(?:bcd|bc|b|(?:))` equivalent to `a(?:bcd|bc|b|)`

Reply:
Yes.  I was just trying to be explicit about the "empty" slot.  Want me to change it?

Reply:
or test both?

Reply:
> or test both?

We already have tests that all non-capturing groups are eliminated.